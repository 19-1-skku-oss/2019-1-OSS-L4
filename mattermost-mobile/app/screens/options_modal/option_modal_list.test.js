// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import OptionModalListIOS from './options_modal_list.ios';
import OptionModalListAndroid from './options_modal_list.android';

describe('OptionModalList', () => {
    const baseProps = {
        items: [{
            action: jest.fn(),
            text: {
                id: 'mobile.file_upload.camera',
                defaultMessage: 'Take Photo or Video',
            },
            icon: 'camera',
        }, {
            action: jest.fn(),
            text: {
                id: 'mobile.file_upload.library',
                defaultMessage: 'Photo Library',
            },
            icon: 'photo',
        }],
        onCancelPress: jest.fn(),
        title: 'test',
    };

    test('should match snapshot for iOS', async () => {
        const wrapper = shallow(
            <OptionModalListIOS {...baseProps}/>,
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for Android', async () => {
        const wrapper = shallow(
            <OptionModalListAndroid {...baseProps}/>,
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
