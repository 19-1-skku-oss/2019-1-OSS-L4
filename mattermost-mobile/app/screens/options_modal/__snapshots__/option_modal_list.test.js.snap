// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionModalList should match snapshot for Android 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "white",
        "borderRadius": 2,
        "marginHorizontal": 30,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "rgba(0, 0, 0, 0.1)",
          "borderBottomWidth": 1,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "padding": 15,
          }
        }
      >
        <FormattedText
          defaultMessage="Take Photo or Video"
          id="mobile.file_upload.camera"
          style={
            Array [
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 16,
              },
              undefined,
            ]
          }
        />
        <Icon
          allowFontScaling={false}
          name="camera"
          size={18}
          style={
            Object {
              "color": "#7f8180",
            }
          }
        />
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(0, 0, 0, 0.1)",
          "borderBottomWidth": 1,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "padding": 15,
          }
        }
      >
        <FormattedText
          defaultMessage="Photo Library"
          id="mobile.file_upload.library"
          style={
            Array [
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 16,
              },
              undefined,
            ]
          }
        />
        <Icon
          allowFontScaling={false}
          name="photo"
          size={18}
          style={
            Object {
              "color": "#7f8180",
            }
          }
        />
      </TouchableOpacity>
    </View>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 15,
        }
      }
    >
      <FormattedText
        defaultMessage="Cancel"
        id="channel_modal.cancel"
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "fontSize": 16,
          }
        }
      />
    </TouchableOpacity>
  </View>
</View>
`;

exports[`OptionModalList should match snapshot for iOS 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-end",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "white",
        "borderRadius": 12,
        "marginBottom": 20,
        "marginHorizontal": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "padding": 15,
          },
          Object {
            "borderBottomColor": "rgba(0, 0, 0, 0.1)",
            "borderBottomWidth": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#7f8180",
            "flex": 1,
            "textAlign": "center",
          }
        }
      >
        test
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "rgba(0, 0, 0, 0.1)",
            "borderBottomWidth": 1,
          },
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "padding": 15,
          }
        }
      >
        <FormattedText
          defaultMessage="Take Photo or Video"
          id="mobile.file_upload.camera"
          style={
            Array [
              Object {
                "color": "#4E8ACC",
                "flex": 1,
                "fontSize": 20,
              },
              undefined,
              false,
            ]
          }
        />
        <Icon
          allowFontScaling={false}
          name="camera"
          size={18}
          style={
            Object {
              "color": "#4E8ACC",
            }
          }
        />
      </TouchableOpacity>
    </View>
    <View
      style={
        Array [
          false,
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "padding": 15,
          }
        }
      >
        <FormattedText
          defaultMessage="Photo Library"
          id="mobile.file_upload.library"
          style={
            Array [
              Object {
                "color": "#4E8ACC",
                "flex": 1,
                "fontSize": 20,
              },
              undefined,
              false,
            ]
          }
        />
        <Icon
          allowFontScaling={false}
          name="photo"
          size={18}
          style={
            Object {
              "color": "#4E8ACC",
            }
          }
        />
      </TouchableOpacity>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "white",
        "borderRadius": 12,
        "marginBottom": 20,
        "marginHorizontal": 20,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 15,
        }
      }
    >
      <FormattedText
        defaultMessage="Cancel"
        id="channel_modal.cancel"
        style={
          Object {
            "color": "#CC3239",
            "flex": 1,
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      />
    </TouchableOpacity>
  </View>
</View>
`;
